app/code/community/Bolt/Boltpay/Model/Payment.php

```diff
@@ -478,7 +480,7 @@ public function handleTransactionUpdate(Mage_Payment_Model_Info $payment, $newTr
                 //Mage::log(sprintf("Valid next states from %s: %s", $prevTransactionStatus, implode(",",$validNextStatuses)), null, 'bolt.log');
 
                 if (!in_array($newTransactionStatus, $validNextStatuses)) {
```

Comment:
I mentioned in the comment above that this is very old code that I saw to update instead of re-design.  Absolutely, Magento's order status can be used by making the updates described above.